x/lightclient/ante/ibc_msg_update_client.go

```diff
@@ -0,0 +1,96 @@
+	"github.com/dymensionxyz/dymension/v3/x/lightclient/types"
+)
+
+func (i IBCMessagesDecorator) HandleMsgUpdateClient(ctx sdk.Context, msg *ibcclienttypes.MsgUpdateClient) error {
+	clientState, found := i.ibcClientKeeper.GetClientState(ctx, msg.ClientId)
+	if !found {
+		return nil
+	}
+	// Only continue if the client is a tendermint client as rollapp only supports tendermint clients as canonical clients
+	if clientState.ClientType() == exported.Tendermint {
+		// Cast client state to tendermint client state - we need this to get the chain id(rollapp id)
+		tmClientState, ok := clientState.(*ibctm.ClientState)
+		if !ok {
+			return nil
+		}
+		// Check if the client is the canonical client for the rollapp
+		rollappID := tmClientState.ChainId
+		canonicalClient, found := i.lightClientKeeper.GetCanonicalClient(ctx, rollappID)
+		if !found || canonicalClient != msg.ClientId {
+			return nil // The client is not a rollapp's canonical client. Continue with default behaviour.
+		}
+		clientMessage, err := ibcclienttypes.UnpackClientMessage(msg.ClientMessage)
+		if err != nil {
+			return nil
+		}
+		header, ok := clientMessage.(*ibctm.Header)
+		if !ok {
+			return nil
+		}
+		// Check if there are existing block descriptors for the given height of client state
+		height := clientState.GetLatestHeight()
+		stateInfo, err := i.rollappKeeper.FindStateInfoByHeight(ctx, rollappID, height.GetRevisionHeight())
+		if err != nil {
+			// No BDs found for given height.
+			// Will accept the update optimistically
+			// But also save the blockProposer address with the height for future verification
+			blockProposer := header.Header.ProposerAddress
+			i.lightClientKeeper.SetConsensusStateSigner(ctx, msg.ClientId, height.GetRevisionHeight(), string(blockProposer))
+			return nil
+		}
+		bd := stateInfo.GetBDs().BD[height.GetRevisionHeight()-stateInfo.GetStartHeight()]
+
+		ibcState := types.IBCState{
+			Root:               header.Header.GetAppHash(),
+			Height:             height.GetRevisionHeight(),
+			Validator:          header.Header.ProposerAddress,
+			NextValidatorsHash: header.Header.NextValidatorsHash,
+			Timestamp:          header.Header.Time,
+		}
+		sequencerPubKey, err := i.lightClientKeeper.GetTmPubkeyAsBytes(ctx, stateInfo.Sequencer)
+		if err != nil {
+			return err
+		}
+		rollappState := types.RollappState{
+			BlockSequencer:  sequencerPubKey,
+			BlockDescriptor: bd,
+		}
+		// Check that BD for next block exists in same stateinfo
+		if height.GetRevisionHeight()-stateInfo.GetStartHeight()+1 < uint64(len(stateInfo.GetBDs().BD)) {
+			rollappState.NextBlockSequencer = sequencerPubKey
+			rollappState.NextBlockDescriptor = stateInfo.GetBDs().BD[height.GetRevisionHeight()-stateInfo.GetStartHeight()+1]
+		} else {
+			// next BD does not exist in this state info, check the next state info
+			nextStateInfo, found := i.rollappKeeper.GetStateInfo(ctx, rollappID, stateInfo.GetIndex().Index+1)
```

Comment:
To be able to get the sequencer address. we will need the entire stateinfo object. so just fetching blockdescriptors by height wont be enough. 
While I could do
`func (k rollappKeeper) FindBlockDescriptorByHeight(ctx sdk.Context, rollappId string, height uint64) (bd types.BlockDescriptor, seqAddr string, err error)`
this would require two binary searches everytime. When in some cases, both bd(h) and bd(h+1) might exist in same stateinfo and its more efficient to lookup there.